src/main/java/com/sourcegraph/cody/SubscriptionTabPanel.kt

```diff
@@ -0,0 +1,106 @@
+package com.sourcegraph.cody
+
+import com.intellij.ide.BrowserUtil
+import com.intellij.ide.ui.LafManagerListener
+import com.intellij.ide.ui.laf.darcula.ui.DarculaButtonUI
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.ui.dsl.builder.panel
+import com.intellij.util.ui.JBUI
+import com.sourcegraph.common.CodyBundle
+import com.sourcegraph.common.UpgradeToCodyProNotification
+import com.sourcegraph.config.ConfigUtil
+import com.sourcegraph.config.ThemeUtil
+import java.awt.BorderLayout
+import javax.swing.JPanel
+import javax.swing.border.EmptyBorder
+
+class SubscriptionTabPanel : JPanel() {
+
+  private var isCurrentUserPro: Boolean? = null
+  private var chatLimitError = UpgradeToCodyProNotification.chatRateLimitError.get()
+  private var autocompleteLimitError = UpgradeToCodyProNotification.autocompleteRateLimitError.get()
+
+  init {
+    layout = BorderLayout()
+    border = EmptyBorder(JBUI.insets(4))
+    if (chatLimitError != null || autocompleteLimitError != null) {
+      add(createRateLimitPanel(), BorderLayout.PAGE_START)
+    }
+    add(createCenterPanel())
+    ApplicationManager.getApplication()
+        .messageBus
+        .connect()
+        .subscribe(
+            LafManagerListener.TOPIC,
+            LafManagerListener {
+              ApplicationManager.getApplication().executeOnPooledThread { update(isCurrentUserPro) }
```

Comment:
there is no need to call it on the pooled thread + annotate `update` method as `@RequiresEdt`